### ISSUE TITLE
it's a bug in code?

### ISSUE THREAD

**Issue by kellenf** [open-mmlab/mmdetection#4892]
lass BBoxTestMixin(object):
    """Mixin class for test time augmentation of bboxes."""

    def merge_aug_bboxes(self, aug_bboxes, aug_scores, img_metas):
        """Merge augmented detection bboxes and scores.
        Args:
            aug_bboxes (list[Tensor]): shape (n, 4*#class)
            aug_scores (list[Tensor] or None): shape (n, #class)
            img_shapes (list[Tensor]): shape (3, ).
        Returns:
            tuple: (bboxes, scores)
        """
        recovered_bboxes = []
        for bboxes, img_info in zip(aug_bboxes, img_metas):
            img_shape = img_info[0]['img_shape']
            scale_factor = img_info[0]['scale_factor']
            flip = img_info[0]['flip']
            flip_direction = img_info[0]['flip_direction']
            bboxes = bbox_mapping_back(bboxes, img_shape, scale_factor, flip,
                                       flip_direction)
            recovered_bboxes.append(bboxes)
        bboxes = torch.cat(recovered_bboxes, dim=0)
        if aug_scores is None:
            return bboxes
        else:
            scores = torch.cat(aug_scores, dim=0)
            return bboxes, scores
in this code it use  img_shape, it should be pad_shape right?

**Comment by hhaAndroid** [comment]
Hi, @kellenf  This is correct because the pad is in the lower right way.

**Comment by kellenf** [comment]
because it's lower right way, so in flip it's not right use img_shape to flip bboxes, right?

**Comment by hhaAndroid** [comment]
> because it's lower right way, so in flip it's not right use img_shape to flip bboxes, right?

That's not what I meant. In the lower right pad style, no need to remove the pad pixels first.
